nni/experiment/config/exp_config.py

```diff
@@ -129,7 +127,7 @@ def _validate_canonical(self):
         space_cnt = (self.search_space is not None) + (self.search_space_file is not None)
         if self.use_annotation and space_cnt != 0:
             raise ValueError('ExperimentConfig: search space must not be set when annotation is enabled')
-        if not self.use_annotation and space_cnt != 1:
+        if not self.use_annotation and space_cnt < 1:
```

Comment:
why we allow `space_cnt=2`?

Reply:
When user specifies `seachSpaceFile`, NNI converts it to `searchSpace` (required by NNI manager). In previous version NNI removes `searchSpaceFile` during conversion, and in this implementation it gets kept. When resuming an experiment, the search space has already been converted and the check will fail.
So there are 3 approaches:
  1. don't store `searchSpaceFile` (user can't view the path in web UI)
  2. don't check it (use can set both `searchSpace` and `searchSpaceFile` and the latter is silently ignored)
  3. don't validate config for resumed experiment (the logic is more complex and might introduce potential problem)
I think it is very unlikely for users to set both `searchSpace` and `searchSpaceFile`, and therefore prefer the second approach.

Reply:
got it